### PR TITLE
Adapter Name Case Insensitive: dealTier

### PR DESCRIPTION
**Pull request by guscarreon**
Function `ReadDealTiersFromImp` unmarshals the `Imp.Ext` and creates a map of bidder names to deal tier info. This pull request normalizes the keys of this entries in this map in order to compare to the normalized bidder names and apply deal tiers if needed.

### CODE REVIEW

**Review by AlexBVolcy — DISMISSED** [review]

File: openrtb_ext/deal_tier.go [hunk 1]
```diff
@@ -39,7 +39,9 @@ func ReadDealTiersFromImp(imp openrtb2.Imp) (DealTierBidderMap, error) {
 	}
 	for bidder, param := range impPrebidExt.Prebid.Bidders {
 		if param.DealTier != nil {
-			dealTiers[BidderName(bidder)] = *param.DealTier
+			if bidderNormalized, bidderFound := NormalizeBidderName(bidder); bidderFound {
+				dealTiers[bidderNormalized] = *param.DealTier
+			}
```
Comment (bsardo) [inline openrtb_ext/deal_tier.go:44]: The deal tier map being created here from the original request has keys that are the normalized bidder names. This map is then used in `applyDealSupport` but that function does not appear to alter the bidder name in `winningBidsByBidder` --> `topBidsPerImp` prior to checking if the bidder has any deal tiers in the deal tier map. Is a change needed in `applyDealSupport` as well?

Comment (guscarreon) [reply openrtb_ext/deal_tier.go:43]: Good catch. Modified.

File: exchange/exchange.go [hunk 2]
```diff
@@ -575,15 +575,18 @@ func applyDealSupport(bidRequest *openrtb2.BidRequest, auc *auction, bidCategory
 	for impID, topBidsPerImp := range auc.winningBidsByBidder {
 		impDeal := impDealMap[impID]
 		for bidder, topBidsPerBidder := range topBidsPerImp {
-			maxBid := bidsToUpdate(multiBid, bidder.String())
-
+			bidderNormalized, bidderFound := openrtb_ext.NormalizeBidderName(bidder.String())
+			if !bidderFound {
+				continue
```
Comment (AlexBVolcy) [inline exchange/exchange.go:580]: This line isn't covered by tests, do you think you can add a test where the bidder is not found from the `NormalizeBidderName` call?

**Review by AlexBVolcy — COMMENTED** [review]
This looks good, just one test coverage comment.

File: exchange/exchange.go (hunk 2 above)
Comment (guscarreon) [reply exchange/exchange.go:580]: Done. Also refactored `TestApplyDealSupport` a little bit.

**Review by AlexBVolcy — APPROVED** [review]
LGTM

**Review by VeronikaSolovei9 — APPROVED** [review]
Local testing looks good, approved

**Comment by VeronikaSolovei9** [comment]
While testing this PR I found that deals don't work with stored bid responses. This happens because impressions with stored bid response are removed from bidRequest and `func getDealTiers` doesn't return `impDealMap` with data for this impression. I'm not sure if this is something that has to be fixed, just want to note about this behavior.

**Pull request merged**